src/lib/arch/riscv/Instruction_execute.cc

```diff
@@ -0,0 +1,837 @@
+
+#include <cmath>
+#include <tuple>
+
+#include "InstructionMetadata.hh"
+#include "simeng/arch/riscv/Instruction.hh"
+
+namespace simeng {
+namespace arch {
+namespace riscv {
+
+/** Multiply unsigned `a` and unsigned `b`, and return the high 64 bits of the
+ * result. https://stackoverflow.com/a/28904636 */
+uint64_t mulhiuu(uint64_t a, uint64_t b) {
+  uint64_t a_lo = (uint32_t)a;
+  uint64_t a_hi = a >> 32;
+  uint64_t b_lo = (uint32_t)b;
+  uint64_t b_hi = b >> 32;
+
+  uint64_t a_x_b_hi = a_hi * b_hi;
+  uint64_t a_x_b_mid = a_hi * b_lo;
+  uint64_t b_x_a_mid = b_hi * a_lo;
+  uint64_t a_x_b_lo = a_lo * b_lo;
+
+  uint64_t carry_bit = ((uint64_t)(uint32_t)a_x_b_mid +
+                        (uint64_t)(uint32_t)b_x_a_mid + (a_x_b_lo >> 32)) >>
+                       32;
+
+  uint64_t multhi =
+      a_x_b_hi + (a_x_b_mid >> 32) + (b_x_a_mid >> 32) + carry_bit;
+
+  return multhi;
+}
+
+/** Multiply signed `a` and signed `b`, and return the high 64 bits of the
+ * result. */
+uint64_t mulhiss(int64_t a, int64_t b) {
```

Comment:
These are placeholders for the two instructions that currently don't have an implementation in the M extension. We currently have a problem in both the AArch64 and RISC-V implementations where we can't correctly compute the high bits of a multiply between a signed and unsigned number or two signed numbers as per SE-149.

The comments remain as a reminder to implement these two instructions as these are the only two not implemented for this extension. It could be cleaner to remove the helper functions, but I want to keep the commented case's. What do you suggest to keep it clean?

Reply:
> "The comments remain as a reminder to implement these two instructions as these are the only two not implemented for this extension."
I don't have a strong preference, but if you feel that it would be easier to track these here and in Jira, then I'm happy with it.
@jj16791 @FinnWilkinson, any thoughts on this?

Reply:
I think leaving them commented with a TODO is fine. Serves as a reminder to implement this and will make debugging simpler for any issues around these instructions

Reply:
As long as we don't upstream the commented-out TODOs into `main`, I'm fine with leaving it as is.